mmf_cli/inference.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3 -u
+# Copyright (c) Facebook, Inc. and its affiliates.
+import argparse
+import typing
+
+from mmf.utils.flags import flags
+from mmf.utils.inference import Inference
+from omegaconf import OmegaConf
+
+
+def construct_config(opts: typing.List[str]):
+    config = OmegaConf.create()
+    opt_values = [opt.split("=", maxsplit=1) for opt in opts]
```

Comment:
Not necessary but I think you should refactor out the logic in configuration as there is many more cases that you need to handle that we have already done in configuration opts logic parser.

Reply:
I tried to use the configuration logic but I came across a few problems: configuration can only take fields that are used in the default config, i.e. it cannot take new fields like `checkpoint` or `checkpoint_path` as `checkpoint` is initialized to something else (a dictconfig struct) and `checkpoint_path` is not included in the default config so it just throws an error. The other problem is the configuration also _requires_ certain fields that inference does not require like `datasets`. The way to work around that is to just programmatically include a random dataset but I figured it would just be easier to do this

Reply:
To deal with the dataset requirement, just modify configuration.py:

```
class Configuration:
    def __init__(self, args=None, default_only=False, load_dataset=True):
...
other_configs = self._build_other_configs(load_dataset=load_dataset)
...
def _build_other_configs(self, load_dataset=True):
        opts_config = self._build_opt_list(self.args.opts)
        user_config = self._build_user_config(opts_config)

        self._opts_config = opts_config
        self._user_config = user_config

        self.import_user_dir()
        model_config = self._build_model_config(opts_config)
        args_overrides = self._build_demjson_config(self.args.config_override)
        other_configs = None
        if load_dataset:
            dataset_config = self._build_dataset_config(opts_config)
            other_configs = OmegaConf.merge(
                model_config, dataset_config, user_config, args_overrides
            )
        else:
            other_configs = OmegaConf.merge(
                model_config, user_config, args_overrides
            )
            

        return other_configs

```


I'm dealing with this issue too as I am using the models for inference, this was the simplest way to deal with it.